src/main/java/com/iexec/worker/tee/pre/PreComputeAppRunner.java

```diff
@@ -16,31 +16,64 @@
 
 package com.iexec.worker.tee.pre;
 
-import com.iexec.common.precompute.PreComputeExitCode;
+import com.iexec.common.replicate.ReplicateStatusCause;
+import com.iexec.common.utils.FeignBuilder;
+import com.iexec.common.worker.api.ExitMessage;
+import com.iexec.worker.tee.pre.worker.WorkerApiClient;
+import feign.FeignException;
+import feign.Logger;
 import lombok.extern.slf4j.Slf4j;
 
+import static com.iexec.common.utils.IexecEnvUtils.IEXEC_TASK_ID;
+import static com.iexec.worker.tee.pre.PreComputeArgs.getEnvVar;
+
 @Slf4j
 public class PreComputeAppRunner {
 
+    public static final String WORKER_HOST = "worker:13100";
+
+    private PreComputeAppRunner() {
+        throw new UnsupportedOperationException();
+    }
+
     /**
      * Run PreComputeApp and handle possible exceptions.
+     * Exits:
+     * - 0: Success
+     * - 1: Failure; Reported cause (known or unknown)
+     * - 2: Failure; Unreported cause since reporting issue
+     * - 3: Failure; Unreported cause since missing taskID context
      */
     public static void start() {
         log.info("TEE pre-compute started");
-        int exitCode = PreComputeExitCode.SUCCESS.value();
+        ReplicateStatusCause exitCause = ReplicateStatusCause.PRE_COMPUTE_UNKNOWN_FAILED;
+        String chainTaskId = getEnvVar(IEXEC_TASK_ID);
```

Comment:
I would prefer something like:
```
        try {
            taskId = EnvUtils.getEnvVarOrThrow(RESULT_TASK_ID, PRE_COMPUTE_MISSING_TASK_ID);
        } catch (PostComputeException e) {
            log.error("TEE pre-compute cannot go further without taskID context");
            System.exit(3);
        }
```

with `getEnvVarOrThrow` defined as:
```
    public static String getEnvVarOrThrow(String envVarName, ReplicateStatusCause statusCauseIfMissing) throws PostComputeException {
        String envVar = System.getenv(envVarName);
        if (envVar == null || envVar.isEmpty()) {
            log.error("Required env var is empty [name:{}]", envVarName);
            throw new PreComputeException(statusCauseIfMissing);
        }
        return envVar;
    }
```

IMHO, throwing an exception when a required arg is missing is better than returning an empty String.